### PR TITLE
Get nearest graphics

### LINKED ISSUE #514: Plot area nearest graphics method
A method on plot area that returns the nearest n graphics could be useful. Just get world bounding sphere of all graphics and return closest in order.. Will need to see if this is fast enough. We could cache the world bounding spheres and update them whenever graphic data, offset or rotation changes.

### PR DESCRIPTION
**Pull request by kushalkolar**
closes #514 

WIP, need to finish edge method

```python
from itertools import product
import numpy as np
import fastplotlib as fpl


def make_circle(center, radius: float, n_points: int = 75) -> np.ndarray:
    theta = np.linspace(0, 2 * np.pi, n_points)
    xs = radius * np.sin(theta)
    ys = radius * np.cos(theta)

    return np.column_stack([xs, ys]) + center


spatial_dims = (100, 100)

circles = list()
for center in product(range(0, spatial_dims[0], 15), range(0, spatial_dims[1], 15)):
    circles.append(make_circle(center, 5, n_points=75))

fig = fpl.Figure()

lines = fig[0, 0].add_line_collection(circles, colors="r", thickness=5)


def highlight_nearest(ev):
    lines.colors = "r"
    
    xy = fig[0, 0].map_screen_to_world(ev)[:-1]
    nearest = fig[0, 0].get_nearest_graphics(xy, method="center", subset=lines)
    
    nearest[0].colors = "w"

fig.renderer.add_event_handler(highlight_nearest, "pointer_move")

fig.show()
```

https://github.com/fastplotlib/fastplotlib/assets/9403332/daf2842c-379b-461b-a07a-73648ca30ab9

### CODE REVIEW

**Comment by kushalkolar** [comment]
@clewis7 @apasarkar what about just skipping the edge method, using the bbox is actually useless for many real use cases like this where there is lots of overlap :upside_down_face: 

image

**Comment by clewis7** [comment]
This looks amazing!! I am okay with skipping edge. When I think about doing something like clicking a contour I would be clicking near the center...

**Comment by kushalkolar** [comment]
I think I'm going to make this a function in `fastplotlib.utils`

**Comment by kushalkolar** [comment]
@clewis7 ready for review!

**Review by clewis7 — COMMENTED** [review]
See question...

File: fastplotlib/utils/_plot_helpers.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    tuple[Graphic]
+        nearest graphics to ``pos`` in order
+
+    """
+
+    if isinstance(graphics, GraphicCollection):
+        graphics = graphics.graphics
+
+    if not all(isinstance(g, Graphic) for g in graphics):
+        raise TypeError("all elements of `graphics` must be Graphic objects")
```
[the remark this answers is not in the document]
Comment (kushalkolar) [reply fastplotlib/utils/_plot_helpers.py:37]: We leave that to the user, so it's just a very simple function unaware of plot areas

**Review by clewis7 — APPROVED** [review]

**Pull request merged**